universal-login-commons/test/integration/ethereum/validators/EnoughTokenValidator.ts

```diff
@@ -0,0 +1,90 @@
+import {expect} from 'chai';
+import {hasEnoughToken, EnoughTokenValidator} from '../../../../lib/integration/ethereum/validators/EnoughTokenValidator';
+import {utils, providers, Wallet, Contract} from 'ethers';
+import {ETHER_NATIVE_TOKEN, EMPTY_DATA} from '../../../../lib/core/constants/constants';
+import {TEST_ACCOUNT_ADDRESS, TEST_CONTRACT_ADDRESS, TEST_PRIVATE_KEY} from '../../../../lib/core/constants/test';
+import {deployContract, getWallets, createMockProvider} from 'ethereum-waffle';
+import MockToken from '../../../fixtures/MockToken.json';
+import {SignedMessage, calculateMessageSignature, BalanceChecker} from '../../../../lib';
+
+describe('INT: EnoughTokenValidator', () => {
+  const provider: providers.Provider = createMockProvider();
+  let wallet: Wallet;
+  const validator = new EnoughTokenValidator(provider);
+  const balanceChecker = new BalanceChecker(provider);
+  const gasLimit = 1000000;
+  const gasPrice = '2';
+  let token: Contract;
+  const walletBalance = utils.parseEther('2');
+  let signedMessage: SignedMessage;
+
+  before(async () => {
+    [wallet] = await getWallets(provider);
+    token = await deployContract(wallet, MockToken);
+    await wallet.sendTransaction({to: TEST_CONTRACT_ADDRESS, value: walletBalance});
+    await token.transfer(TEST_CONTRACT_ADDRESS, utils.parseEther('1'));
+    const message = {
+      from: TEST_CONTRACT_ADDRESS,
+      to: TEST_ACCOUNT_ADDRESS,
+      value: utils.parseEther('1.0'),
+      data: EMPTY_DATA,
+      nonce: 0,
+      gasPrice,
+      gasLimitExecution: utils.bigNumberify(190000),
+      gasData: utils.bigNumberify(10000),
+      gasToken: token.address,
+    };
+    signedMessage = {
+      ...message,
+      signature: calculateMessageSignature(TEST_PRIVATE_KEY, message),
+    };
+  });
+
+  it('successfully pass the validation', async () => {
+    await expect(validator.validate(signedMessage)).to.not.be.rejected;
+  });
+
+  it('passes when not enough gas', async () => {
+    await expect(validator.validate({...signedMessage, gasLimitExecution: 100, gasData: 1000})).to.be.eventually.fulfilled;
+  });
+
+  it('throws when not enough tokens', async () => {
+    await expect(validator.validate({...signedMessage, gasLimitExecution: utils.parseEther('2.0')})).to.be.eventually.rejectedWith('Not enough tokens');
+  });
+
+  describe('hasEnoughToken', async () => {
```

Comment:
Move `hasEnoughToken` tests to separate file

Reply:
Why? This function  is directly connected with Vaildator

Reply:
Yes, it is but:
- name of the file tells that here is the test of EnoughTokenValidator, not tests of connected functions.
- the file growth too much -> it is really hard to read it.

Reply:
I think that it is not neccessary because this method exported only for test purposes relating to Validator

Reply:
I agree with @JustynaBroniszewska

Reply:
Done.